File: alg_selection_sort.java
/******************************************************************************
# l0m1s
# lukes1582@gmail.com
algoritmo selection sort sviluppato per Java
*******************************************************************************/
public class alg_selection_sort {
	public static void main(String[] args) {
        int arrayB[]={5,9,2,3,1,0,4};

    for(int iVar = 0; iVar<arrayB.length; iVar++){
        System.out.print(Integer.toString(arrayB[iVar])+" ");
    }
    System.out.println();

    for(int jVar=0; jVar<arrayB.length-1;jVar++){ // dove 6 e' il valore della lunghezza dell'array meno 1
        int indexMin= jVar;

        for(int kVar= jVar+1;kVar<arrayB.length;kVar++){ // dove 7 è la lunghezza dell'array
            if(arrayB[indexMin]>arrayB[kVar]){
                indexMin=kVar;
            }
        }
        int temp=arrayB[jVar];
        arrayB[jVar]=arrayB[indexMin];
        arrayB[indexMin]=temp;
        
        for(int wVar=0;wVar<7;wVar++){
            System.out.print(Integer.toString(arrayB[wVar])+" ");
        }
        System.out.println();
	
    }
		

	}

}

File: alg_selection_sort.py
#!/usr/bin/env python3

"""
l0m1s
lukes1582@gmail.com
algoritmo selection sort sviluppato per Python
"""

arr = [5,4,3,1,2,11,9,8,0]
print("Array in origine ")
print(arr)
print ("Lunghezza dell'array "+ str(len(arr)))
print(50*"x")

for i in range(len(arr)):
    min_idx = i
    for j in range(i+1, len(arr)):
        if arr[min_idx] > arr[j]:
            min_idx = j
    print(arr)
    arr[i], arr[min_idx] = arr[min_idx], arr[i]
    
